### ISSUE TITLE
OTP master macOS ssl_local_echo test failure

### ISSUE THREAD

**Issue by essen** [ninenines/ranch#351]
Such as https://github.com/ninenines/ranch/actions/runs/11705826701/job/32603377194

```
=== Ended at 2024-11-06 15:13:12
=== Location: [{acceptor_SUITE,do_ssl_local_echo,[855](https://github.com/ninenines/ranch/issues/acceptor_suite.src.html#855)},
              {test_server,ts_tc,1794},
              {test_server,run_test_case_eval1,1303},
              {test_server,run_test_case_eval,1235}]
=== === Reason: no match of right hand side value {error,badarg}
  in function  acceptor_SUITE:do_ssl_local_echo/0 (test/acceptor_SUITE.erl, line 855)
  in call from test_server:ts_tc/3 (test_server.erl, line 1794)
  in call from test_server:run_test_case_eval1/6 (test_server.erl, line 1303)
  in call from test_server:run_test_case_eval/9 (test_server.erl, line 1235)
```

Not sure where it goes wrong exactly, will need to investigate.

cc @juhlig

**Comment by juhlig** [comment]
cc @Maria-12648430

**Comment by Maria-12648430** [comment]
There was a bug in `ssl` in OTP 27, such that local sockets didn't work with `ssl` at all, any OS. That was fixed, though: erlang/otp#8465. So this is very likely a different problem. I can't help there, sorry, because no macOS :woman_shrugging:

**Comment by essen** [comment]
This is indeed fixed in current master. Closing, thanks!
